src/utils/domainTraverse.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  DomainHierarchy,
+  DomainReader,
+  DomainTransactionFactory,
+  IRoleDefinition,
+  ResolverContractType,
+} from '@energyweb/iam-contracts';
+import { providers } from 'ethers';
+import { Methods } from '@ew-did-registry/did';
+import { Signer } from 'ethers';
+import { isValidDID } from './did';
+import { IDefinition } from '../modules/domains';
+import { ENSRegistry__factory } from '../../ethers/factories/ENSRegistry__factory';
+import { labelhash, namehash } from './ensHash';
+import { ChainId, chainConfigs } from '../config';
+
+const { JsonRpcProvider } = providers;
+
+/**
+ * @description - Checks that role issuers of all roles under `rootDomain` contains method-specific-id and adds it if missing
+ * `signer` must own `rootDomain` on `targetChain`
+ */
+export const updateLegacyIssuers = async (
+  rootDomain: string,
+  signer: Signer,
+  chainId: ChainId,
+  dryRun = true
+) => {
+  const { transactionFactory, chainName, provider, domainHierarchy, domainReader, ensRegistry } =
+    await initDomains(signer, chainId);
+
+  let updated = new Array<any>();
+  const update = async (domain: string) => {
+    const domainHash = namehash(domain);
+    let def: IDefinition;
+    def = await domainReader.read({ node: domainHash });
+    if (DomainReader.isRoleDefinition(def)) {
+
+      const {
+        issuer: { did },
+        version,
+      } = def;
+      const updatedDef: IRoleDefinition = {
+        ...def,
+        issuer: {
+          ...def.issuer,
+          did: did?.map((d) =>
+            isValidDID(d) ? d : `did:${Methods.Erc1056}:${chainName}:${d.split(':').slice(-1)}`
+          ),
+        },
+        version: parseInt(version.toString(), 10),
+      };
+      const updateTx = transactionFactory.editDomain({
+        domain,
+        domainDefinition: updatedDef,
+      });
```

Comment:
Instead of just `editDomain`, I think there is some additional work that needs to be done to change ENSRegistry to point to new resolver and set reverse name. See https://github.com/energywebfoundation/iam-client-lib/blob/eae6b169f5c9e960655cd38a6f8d414297bd55d1/src/modules/domains/domains.service.ts#L899